doc/api/module.md

```diff
@@ -1048,6 +1048,92 @@ returned object contains the following keys:
 * columnNumber: {number} The 1-indexed columnNumber of the
   corresponding call site in the original source
 
+## TypeScript support
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+The flag [`--experimental-strip-types`][] enables Node.js to run TypeScript
+files that contain only type annotations. Such files contain no TypeScript
+features that require transformation, such as enums or namespaces. Node.js will
+replace inline type annotations with whitespace, and no type checking is
+performed. TypeScript features that depend on settings within `tsconfig.json`,
+such as paths or converting newer JavaScript syntax to older standards, are
+intentionally unsupported.
+
+To get fuller TypeScript support, including support for enums and namespaces
+and paths, see <https://nodejs.org/en/learn/getting-started/nodejs-with-typescript>.
+The built-in TypeScript support is designed to be lightweight and as fast as
+possible. By intentionally not supporting syntaxes that require JavaScript code
+generation, and by replacing stripped types with whitespace, Node.js can run
+TypeScript code without the need for source maps and with less overhead.
+
+### Determining module system
+
+Node.js supports both [CommonJS][] and [ES Modules][] syntax in TypeScript
+files. Node.js will not convert from one module system to another; if you want
+your code to run as an ES module, you must use `import` and `export` syntax,
+and if you want your code to run as CommonJS you must use `require` and
+`module.exports`.
+
+* `.ts` files will have their module system determined
+  [the same way as `.js` files.][] To use `import` and `export` syntax, add
+  `"type": "module"` to the nearest parent `package.json`.
+* `.mts` files will always be run as ES modules, similar to `.mjs` files.
+* `.cts` files will always be run as CommonJS modules, similar to `.cjs` files.
+* `.tsx` files are unsupported.
+
+As in JavaScript files, [file extensions are mandatory][] in `import` statements
+and `import()` expressions: `import './file.ts'`, not `import './file'`.
+Because of backward compatibility, file extensions are also mandatory in
+`require()` calls: `require('./file.ts')`, not `require('./file')`, similar to
+how the `.cjs` extension is mandatory in `require` calls in CommonJS files.
+
+The `tsconfig.json` option `allowImportingTsExtensions` will allow the
+TypeScript compiler `tsc` to type-check files with `import` specifiers that
+include the `.ts` extension.
+
+### Unsupported TypeScript features
```

Comment:
I think we should add something like that in this section:

Types must be imported with the `type` keyword:

```ts
import type { Type1, Type2 } from './module.ts';
import { fn, type FnParams } from './fn.ts';
```

Without the `type` keyword, the import would not be stripped and would fail at runtime.

Reply:
done